Adding method to fetch mining pool statistics

schnogz opened the pull request:
This PR addresses issue #23.

Quirks:
In the `getPoolData` method, I had to create a new API Object to point to `https://api.blockchain.info` instead of the default `https://blockchain.info` base url.  If you hit the "normal" base url, you will receive the HTML for this [page](https://blockchain.info/pools).  Please let me if there is a different way to do this.... ¯\\__(ツ)_/¯

Other changes:
- Unit tests for new method
- Added a .editorconfig file to fix spacing/indenting issues between IDE's
- fixed spacing/indenting issues from my last PR
- fixed typo in `exchange/readme.md`

jtormey commented on statistics/index.js:
```diff
@@ -34,3 +35,14 @@ function getChartData(chartType, options) {
 }
+
+function getPoolData(options) {
+  var api = new API('https://api.blockchain.info', { pools : new UrlPattern('/pools?(timespan=:timespan)(:days)(&api_code=:apiCode)') });
```
You can use the existing API object by adding `?format=json` to the url pattern. See [here](https://blockchain.info/pools?format=json&timespan=8days).

jtormey commented on statistics/index.js:
```diff
@@ -34,3 +35,14 @@ function getChartData(chartType, options) {
+
+  options = options || {};
+  return api
+    .request('pools', { apiCode: options.apiCode, timespan: options.timespan, days: options.timespan ? 'days' : '' })
```
Can avoid adding `days` here by changing the url pattern so that it's included with the timespan: `(timespan=:timespan\\days)`

jtormey commented on statistics/index.js:
```diff
@@ -34,3 +35,14 @@ function getChartData(chartType, options) {
+  return api
+    .request('pools', { apiCode: options.apiCode, timespan: options.timespan, days: options.timespan ? 'days' : '' })
+    .then(function (data) {
+      return data || q.reject('Invalid timespan');
```
Is this needed? It looks like our API handles this error: https://blockchain.info/pools?format=json&timespan=11days

schnogz commented:
Addressed your comments with my latest commit.  Sorry that I missed the obvious `?format=json` url pattern and was unaware of `//days` pattern provided by the UrlPattern module.

Thanks for all the help with the PR @jtormey!

jtormey commented:
Thanks for the updates!
